Paco/src/com/pacoapp/paco/sensors/android/RuntimePermissions.java

```diff
@@ -0,0 +1,588 @@
+  public static final String PAYLOAD_PERMISSION_GRANTED = "paco_accessibility_payload_permissiongranted";
+  public static final String PAYLOAD_PERMISSION_USERINITIATED = "paco_accessibility_payload_permissionuserinitiated";
+  public static final String PAYLOAD_PERMISSION_PACKAGES = "paco_accessibility_payload_permissionpackages";
+  public static final String PAYLOAD_PERMISSION_APPNAME = "paco_accessibility_payload_permissionappname";
+  // Number of milliseconds before a permission request is considered stale
+  private static final long PERMISSION_REQUEST_HISTORY_MILLIS = 60000;
+
+  // List of the names of the PERMISSION_GROUPS as they appear in the en_us localization of the
+  // PackageInstaller settings
+  public static final List<String> PERMISSION_SETTINGS_STRINGS = Arrays.asList(
+    "Camera",
+    "Contacts",
+    "Location",
+    "Microphone",
+    "Phone",
+    "Storage",
+    "Body Sensors",
+    "Calendar",
+    "SMS"
+  );
+  // List of the names of the PERMISSION_GROUPS as they appear in the en_us localization of the
+  // runtime permission dialogs. These are mapped to the list of strings in the
+  // PERMISSION_SETTINGS_STRINGS list
+  public static final Map<String, String> PERMISSION_DIALOG_STRINGS = new HashMap<String,String>();;
+  static {
+    PERMISSION_DIALOG_STRINGS.put("take pictures and record video", "Camera");
+    PERMISSION_DIALOG_STRINGS.put("access your contacts", "Contacts");
+    PERMISSION_DIALOG_STRINGS.put("access this device's location", "Location");
+    PERMISSION_DIALOG_STRINGS.put("record audio", "Microphone");
+    PERMISSION_DIALOG_STRINGS.put("make and manage phone calls", "Phone");
+    PERMISSION_DIALOG_STRINGS.put("access photos, media, and files on your device", "Storage");
+    PERMISSION_DIALOG_STRINGS.put("access sensor data about your vital signs", "Body Sensors");
+    PERMISSION_DIALOG_STRINGS.put("access your calendar", "Calendar");
+    PERMISSION_DIALOG_STRINGS.put("send and view SMS messages", "SMS");
+  }
+
+  // Keeps whether the service is connected
+  private static boolean running = false;
+  /**
+   * Used to keep track of which app we are changing settings for. Needed because
+   * AccessibilityEvents will only show us what information is currently being interacted with
+   */
+  private static ArrayList<String> currentlyHandledAppPackageNames;
+  private static String currentlyHandledAppName;
+  private static String currentlyHandledPermission;
+
+  /**
+   * Only used with runtime permission dialogs. Keep the currently requested permission in memory
+   * so we remember it when the user actually clicked allow/deny. This is a queue because
+   * accessibility events might be interleaved (i.e., the next permission request may be triggering
+   * an accessibility event even before the action of the user on the previous request triggers one
+   */
+  private static Queue<EncounteredPermissionRequest> previouslyEncounteredPermissionRequests;
```

Comment:
How does the service deal with getting killed? Does the list of previously encountered permissions need to be persisted and restored across reboots or app updates?